client/nginx.go

```diff
@@ -469,32 +470,54 @@ func (client *NginxClient) DeleteHTTPServer(upstream string, server string) erro
 // UpdateHTTPServers updates the servers of the upstream.
 // Servers that are in the slice, but don't exist in NGINX will be added to NGINX.
 // Servers that aren't in the slice, but exist in NGINX, will be removed from NGINX.
-func (client *NginxClient) UpdateHTTPServers(upstream string, servers []UpstreamServer) ([]UpstreamServer, []UpstreamServer, error) {
+// Servers that are in the slice and exist NGINX, but have different parameters, will be updated.
+func (client *NginxClient) UpdateHTTPServers(upstream string, servers []UpstreamServer) (added []UpstreamServer, deleted []UpstreamServer, updated []UpstreamServer, err error) {
 	serversInNginx, err := client.GetHTTPServers(upstream)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to update servers of %v upstream: %v", upstream, err)
+		return nil, nil, nil, fmt.Errorf("failed to update servers of %v upstream: %v", upstream, err)
 	}
 
-	toAdd, toDelete := determineUpdates(servers, serversInNginx)
+	toAdd, toDelete, toUpdate := determineUpdates(servers, serversInNginx)
 
 	for _, server := range toAdd {
 		err := client.AddHTTPServer(upstream, server)
 		if err != nil {
-			return nil, nil, fmt.Errorf("failed to update servers of %v upstream: %v", upstream, err)
+			return nil, nil, nil, fmt.Errorf("failed to update servers of %v upstream: %v", upstream, err)
 		}
 	}
 
 	for _, server := range toDelete {
 		err := client.DeleteHTTPServer(upstream, server.Server)
 		if err != nil {
-			return nil, nil, fmt.Errorf("failed to update servers of %v upstream: %v", upstream, err)
+			return nil, nil, nil, fmt.Errorf("failed to update servers of %v upstream: %v", upstream, err)
 		}
 	}
 
-	return toAdd, toDelete, nil
+	for _, server := range toUpdate {
+		err := client.UpdateHTTPServer(upstream, server)
+		if err != nil {
+			return nil, nil, nil, fmt.Errorf("failed to update servers of %v upstream: %v", upstream, err)
+		}
+	}
+
+	return toAdd, toDelete, toUpdate, nil
 }
 
-func determineUpdates(updatedServers []UpstreamServer, nginxServers []UpstreamServer) (toAdd []UpstreamServer, toRemove []UpstreamServer) {
+func determineUpdates(updatedServers []UpstreamServer, nginxServers []UpstreamServer) (toAdd []UpstreamServer, toRemove []UpstreamServer, toUpdate []UpstreamServer) {
+	for _, server := range updatedServers {
+		updateFound := false
+		for _, serverNGX := range nginxServers {
+			server.ID = serverNGX.ID
```

Comment:
`I added a suggestion. What happened with the NGINX config files? it seems something happened with the spacing?` Yes

Reply:
> In other parts of the file we ignore the ids by setting it to 0, I suggest do it the same way here:
> `serverNGX.ID = 0` Because server.ID should be 0 as is the zeroValue for int.

Yes, but `server.ID = serverNGX.ID` is required, `UpdateHTTPServers` and `UpdateStreamServer` methods are required the`server.ID` to contruct path `path := fmt.Sprintf("http/upstreams/%v/servers/%v", upstream, server.ID)` and `path := fmt.Sprintf("stream/upstreams/%v/servers/%v", upstream, server.ID)`

`server.ID = 0` is set to zero after path is constructed.

What do you suggest?
Please correct me if anything is missing or incorrect

Reply:
👍